oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java

```diff
@@ -0,0 +1,446 @@
+        for (int i = 1; i < flatFileList.size(); i++) {
+            assertEquals(assetNodeType, startLineType(flatFileList.get(i)));
+        }
+    }
+
+    @Test
+    public void splitFolder() throws IOException, IllegalAccessException {
+        Set<String> splitNodeTypeNames = new HashSet<>(Arrays.asList(
+                "nt:file",
+                "cq:VirtualComponent",
+                "nt:folder",
+                "cq:PollConfigFolder",
+                "cq:ExporterConfigFolder",
+                "cq:ClientLibraryFolder",
+                "cq:ComponentMixin",
+                "cq:ContentSyncConfig",
+                "cq:Component",
+                "sling:OrderedFolder",
+                "sling:Folder",
+                "granite:Component"));
+        File flatFile = new File(classLoader.getResource("complex-split.json").getFile());
+        int expectedSplitSize = 2;
+        FlatFileSplitter splitter = createTestSplitter(flatFile, 0, expectedSplitSize, false, splitNodeTypeNames);
+
+        List<File> flatFileList = splitter.split(false);
+
+        assertTrue(expectedSplitSize <= flatFileList.size());
+        assertEquals(flatFile.length(), getTotalSize(flatFileList));
+        assertEquals(startLineType(flatFile), startLineType(flatFileList.get(0)));
+        String expectedSplitPoint = "/etc|{\"jcr:primaryType\":\"nam:sling:Folder\"}";
+        assertEquals(expectedSplitPoint, startLine(flatFileList.get(1)));
+    }
+
+    @Test
+    public void splitFolderWithCompression() throws IOException, IllegalAccessException {
+        Set<String> splitNodeTypeNames = new HashSet<>(Arrays.asList(
+                "nt:file",
+                "cq:VirtualComponent",
+                "nt:folder",
+                "cq:PollConfigFolder",
+                "cq:ExporterConfigFolder",
+                "cq:ClientLibraryFolder",
+                "cq:ComponentMixin",
+                "cq:ContentSyncConfig",
+                "cq:Component",
+                "sling:OrderedFolder",
+                "sling:Folder",
+                "granite:Component"));
+        File rawFlatFile = new File(classLoader.getResource("complex-split.json").getFile());;
+        File flatFile = temporaryFolder.newFile();
+        compress(rawFlatFile, flatFile);
+        int expectedSplitSize = 3;
+        FlatFileSplitter splitter = createTestSplitter(flatFile, 0, expectedSplitSize, true, splitNodeTypeNames);
+
+        List<File> flatFileList = splitter.split(false);
+        List<File> rawFlatFileList = new ArrayList<>();
+
+        for (File f: flatFileList) {
+            File uf = temporaryFolder.newFile();
+            uncompress(f, uf);
+            rawFlatFileList.add(uf);
+        }
+
+        assertTrue(expectedSplitSize <= flatFileList.size());
+        assertEquals(rawFlatFile.length(), getTotalSize(rawFlatFileList));
+    }
+
+    @Test
+    public void getSplitNodeTypeNames() throws IllegalAccessException {
```

Comment:
no exception is thrown here
```suggestion
    public void getSplitNodeTypeNames() {
```